File: ThirdIteration/java/Priest.java
public class Priest extends Character implements Attribute {


    public Priest(String name) {
        super(name);
        this.name = name;
    }


    @Override
    public void attack(Character character) {
        return;

    }

    @Override
    public void heal(Character character) {

        if (character.health == 100)
            return;

        character.health += Helper.randomNumber(5, 10);;
        character.isAlive();
        character.setHealthto100();


    }


}

File: RPG/src/Priest.kt
import kotlin.random.Random

class Priest(
    name: String,
    health: Int = 100
): Character(name, health) {

    fun attack(anEntity : Entity) {
        super.attack(anEntity, 0)
        println("${anEntity.name} didn't take any damage !")
    }

    fun heal(aCharacter: Character) {
        if (this.isEnemy(aCharacter))
            println("can't heal ${aCharacter.name}, it is an enemy!")
        else
            super.heal(aCharacter, Random.nextInt(5, 11))
    }
}
